conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUndertowRuntime.java

```diff
@@ -132,8 +141,51 @@ public Builder requestArgHandler(RequestArgHandler value) {
             return this;
         }
 
+        @CanIgnoreReturnValue
+        public Builder onSetRequestToken(BiConsumer<HttpServerExchange, Optional<UnverifiedJsonWebToken>> callback) {
+            onSetRequestToken = callback;
+            return this;
+        }
+
         public ConjureUndertowRuntime build() {
             return new ConjureUndertowRuntime(this);
         }
     }
+
+    private static final class OnSetRequestTokenAuthExtractor implements AuthorizationExtractor {
+        private final AuthorizationExtractor delegate;
+        private final BiConsumer<HttpServerExchange, Optional<UnverifiedJsonWebToken>> onSetRequestToken;
+
+        private OnSetRequestTokenAuthExtractor(
+                AuthorizationExtractor delegate,
+                BiConsumer<HttpServerExchange, Optional<UnverifiedJsonWebToken>> onSetRequestToken) {
+            this.delegate = delegate;
+            this.onSetRequestToken = onSetRequestToken;
+        }
+
+        @Override
+        public AuthHeader header(HttpServerExchange exchange) {
+            AuthHeader result = delegate.header(exchange);
+            Optional<UnverifiedJsonWebToken> token = exchange.getAttachment(Attachments.UNVERIFIED_JWT);
+            Preconditions.checkNotNull(token, "jwt in exchange attachment is null");
+            onSetRequestToken.accept(exchange, token);
+            return result;
+        }
+
+        @Override
+        public BearerToken cookie(HttpServerExchange exchange, String cookieName) {
+            BearerToken result = delegate.cookie(exchange, cookieName);
+            Optional<UnverifiedJsonWebToken> token = exchange.getAttachment(Attachments.UNVERIFIED_JWT);
+            Preconditions.checkNotNull(token, "jwt in exchange attachment is null");
+            onSetRequestToken.accept(exchange, token);
+            return result;
+        }
+
+        @Override
+        public void setRequestToken(HttpServerExchange exchange, Optional<UnverifiedJsonWebToken> token) {
+            delegate.setRequestToken(exchange, token);
+            // may throw
+            onSetRequestToken.accept(exchange, token);
```

Comment:
Let's add test coverage including one where the custom handler throws, to ensure this work

Reply:
sure thing, added